### ISSUE TITLE
How to version in-app purchases across multiple iOS reviews?

### ISSUE THREAD

**Issue by mepc36** [j3k0/cordova-plugin-purchase#1300]
### Observed behavior

Hello, I'm a Fovea customer who's wondering this: what is the established pattern for managing iOS in-app purchases across multiple app versions? 

Here's what's happened:

1. Last month, I submitted a new app version using cordova-plugin-purchase, along with 2 in-app purchases.
2. Apple approved it last month.
3. Yesterday I went to submit a new app version.
4. Today, Apple rejected it for this reason: 

```
# Guideline 2.1 - Performance - App Completeness

We found that while you have submitted in-app purchase products for your app, 
the in-app purchase functionality is not present in your binary.

Specifically, Advanced Search for $0.99.

# Next Steps

If you would like to include in-app purchases in your app, 
you will need to upload a new binary that incorporates the in-app 
purchase API to enable users to make a purchase.

Once you revise and resubmit your binary, you will also need to resubmit 
your in-app purchases for review since they are in the Developer Action 
Required state. For each in-app purchase product submitted, please 
be sure to edit the detail information or cancel the request to change 
the detail information for the in-app purchases using App Store Connect.

Alternatively, if you do not want to include in-app purchase products in 
your app, it would be appropriate to remove any unused in-app purchase 
products from App Store Connect.

# Resources

For more information on how to implement in-app purchase in your app, 
please refer to the [In-App Purchase Programming Guide](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Introduction.html).

Learn more about [offering in-app purchases](https://help.apple.com/app-store-connect/#/devb57be10e7) in App Store Connect Help.
```

I searched this error in this GitHub issues, but found only [this](https://github.com/j3k0/cordova-plugin-purchase/issues/705).

### Expected behavior

I thought that iOS would let you attach already-approved in-app purchases when submitting any subsequent, new app version binaries. However, according to the documentation shown in the attached screenshot, Apple you can only submit in-app purchases that are in the "Ready To Submit" state: 

![Screen Shot 2022-04-21 at 5 14 32 PM](https://user-images.githubusercontent.com/16109633/164553400-305c3952-2b2f-40d7-8a31-32f1ec6a37a6.png)

However, adding new IAP's with each new app version would become a real headache, because each new one requires a new product ID, so enabling the app functionality that the I.A.P. buys for a user would quickly get out of hand.

### System Info

"cordova-plugin-purchase": "^11.0.0",

I'm running `@awesome-cordova-plugins/in-app-purchase-2` inside a JavaScript Capacitor app.

*I know this question is not 100% specific to this library, but Apple & Googling is absolutely no help and this community has been super helpful & vibrant in the help. Just looking for someone to maybe unblock me, thank you for reading! 

**TL;DR - How do I version in-app purchases across multiple iOS reviews?**

Thank you for all your hard work on this plugin! Never thought it'd be this easy, and Fovea's UI is intuitive and super easy to use...looking forward to the day when we have to start paying for the Silver plan :)

**Comment by mepc36** [comment]
Resolved & closing.

Versioning purchases is not the problem here; the problem is that Apple does not make it clear how you're supposed to attach already-approved in-app purchases to new binary versions. Crazy as it might seem, literally just writing "Please review this binary with in-app purchases Product ID #123 and Product ID #456" in the "Review Notes" section of the binary submission page fixed this. I found this idea on S.O.F. [here](https://stackoverflow.com/a/44530298).

**Comment by j3k0** [comment]
Sorry for the delay, support email is more reactive. Yeah it looks like Apple's review process and in-app purchases has some rough edges. Sometime you just have to insist in getting someone to take a second look and it will pass...

Wishing you success with your app!
